kubernetes_env/benchmark/benchmark.py

```diff
@@ -139,73 +184,103 @@ def post_request(_):
-                log.error(
-                    "Building filter failed for %s."
-                    " Make sure you give the right path", fd)
-                return util.EXIT_FAILURE
-
-            filter_res = kube_env.refresh_filter(fd)
-            if filter_res != util.EXIT_SUCCESS:
-                log.error(
-                    "Deploying filter failed for %s."
-                    " Make sure you give the right path", fd)
+    for (idx, filter_dir) in enumerate(filter_dirs):
+        log.info("Benchmarking %s", filter_dir)
+        fname = Path(filter_dir).name
+        if filter_dir != "no_filter":
+            res = build_and_deploy_filter(filter_dir)
+            if res != util.EXIT_SUCCESS:
                 return util.EXIT_FAILURE
-
-            # wait for kubernetes set up to finish
-            time.sleep(120)
-            fname = Path(fd).name
             filters.append(fname)
+
         log.info("Warming up...")
         for i in range(10):
             requests.get(url)
-        if custom == "fortio":
+
+        if custom == "locust":
+            if not application:
+                log.error("Provided application does not exists")
+                return util.EXIT_FAILURE
+            log.info("Running locust...")
+            res = run_locust(f"http://{gateway_url}", platform, command_args,
+                             application, fname)
+            if res != util.EXIT_SUCCESS:
+                log.error("Error benchmarking %s application", application)
+                return util.EXIT_FAILURE
+        elif custom == "fortio":
             log.info("Running fortio...")
-            fortio_res = run_fortio(url, platform, request, threads, qps,
-                                    run_time, fname)
+            fortio_res = run_fortio(url, platform, threads, qps, run_time,
+                                    command_args, fname)
             if fortio_res != util.EXIT_SUCCESS:
                 log.error("Error benchmarking for %s", fd)
                 return util.EXIT_FAILURE
         else:
             log.info("Generating load...")
-            burst_res = do_burst(url, platform, request, threads, qps,
-                                 run_time)
+            burst_res = run_loadgen(url, platform, threads, qps, run_time,
+                                    request)
             results.append(burst_res)
 
-    if custom == "fortio":
+    # Plot functions
+    if custom == "locust":
+        locust_df = transform_locust_data(filters, application, path)
+        return plot(locust_df, filters, "", output, "locust")
+    elif custom == "fortio":
         fortio_df, title = transform_fortio_data(filters)
         np.save("fortio", fortio_df)
-        return plot(fortio_df, filters, title, output, fortio=True)
+        return plot(fortio_df, filters, title, output, "fortio")
     else:
         loadgen_df = transform_loadgen_data(filters, results)
         np.save("output", loadgen_df)
```

Comment:
Instead of saving the data as "output" or "fortio", can you save it as "output" + application + time?  That way we have the data saved from different runs.  And then could you also add an option to graph it without rerunning experiments?  I think it would be sufficient to call np.load() and then your plot function.  Then we can examine the data after the fact in a bit more complex ways.